psi4/src/psi4/libmints/zora.cc

```diff
@@ -0,0 +1,263 @@
+    nthreads = Process::environment.get_n_threads();
+#endif
+
+	// Basis function computer on each thread
+	std::vector<std::shared_ptr<BasisFunctions>> pworkers;
+	for (int i = 0; i < nthreads; i++) {
+		auto p_tmp = std::make_shared<BasisFunctions>(primary_, max_points, max_funcs);
+		p_tmp->set_deriv(1);
+		pworkers.push_back(p_tmp);
+	}
+
+	timer_on("Effective Potential");
+	veff_ = std::make_shared<std::unordered_map<int, SharedVector>>();
+	compute_veff();
+	timer_off("Effective Potential");
+
+	timer_on("Scalar Relativistic Kinetic");
+	compute_TSR(pworkers, T_SR);
+	timer_off("Scalar Relativistic Kinetic");
+
+	timer_off("ZORA");
+
+#if ZORADEBUG
+	T_SR->print();
+#endif
+}
+
+void ZORA::compute_veff()
+{
+	// Speed of light in atomic units
+	double C = pc_c_au;
+
+	int nthreads = 1;
+#ifdef _OPENMP
+    nthreads = Process::environment.get_n_threads();
+#endif
+
+	int natoms = molecule_->natom();
+	for (int a = 0; a < natoms; a++) {
+		int Z = molecule_->Z(a);
+		if (Z > 104) throw PSIEXCEPTION("Z too big. Max value 104");
+		auto pos_a = molecule_->xyz(a);
+
+		const double* coef_a  = &coeffs[c_aIndex[Z-1]];
+		const double* alpha_a = &alphas[c_aIndex[Z-1]];
+		int nc_a = c_aIndex[Z] - c_aIndex[Z-1];
+
+#pragma omp parallel for schedule(dynamic) num_threads(nthreads)
+		for (const auto &block : grid_->blocks()) {
+			int npoints = block->npoints();
+			int index = block->index();
+
+			double* x = block->x();
+			double* y = block->y();
+			double* z = block->z();
+
+			auto veff_block = std::make_shared<Vector>(npoints);
+
+			//einsums("i,ip->p", 𝕔[i], erf(α[i]⊗ r[p]))/r[p]
+			for (int p = 0; p < npoints; p++) {
+				double dist = std::hypot(pos_a[0]-x[p], pos_a[1]-y[p], pos_a[2]-z[p]);
+				double outer = 0;
+				for (int i = 0; i < nc_a; i++) {
+					outer += std::erf(dist * alpha_a[i]) * coef_a[i];
+				}
+				outer /= dist;
+				outer -= Z/dist;
+				veff_block->add(p, outer);
+			}
+
+#pragma omp critical
+			{
+				veff_->insert({index, veff_block});
+			}
+		}
+	}
+}
+
+
+//Scalar Relativistic Kinetic Energy Matrix
+void ZORA::compute_TSR(std::vector<std::shared_ptr<BasisFunctions>> pworkers, SharedMatrix &T_SR)
+{
+	// Speed of light in atomic units
+	double C = pc_c_au;
+
+	int nthreads = 1;
+#ifdef _OPENMP
+    nthreads = Process::environment.get_n_threads();
+#endif
+
+#pragma omp parallel for schedule(dynamic) num_threads(nthreads)
+	for (const auto &block : grid_->blocks()) {
+		const auto &bf_map = block->functions_local_to_global();
+		auto local_nbf = bf_map.size();
+		int npoints = block->npoints();
+
+		auto veff_block = veff_->at(block->index());
+
+		int thread = omp_get_thread_num();
+		pworkers[thread]->compute_functions(block);
+		auto phi_x = pworkers[thread]->basis_value("PHI_X");
+		auto phi_y = pworkers[thread]->basis_value("PHI_Y");
+		auto phi_z = pworkers[thread]->basis_value("PHI_Z");
+
+
+		// Preprocess kernel c²/(2c²-veff) * weight
+		double* w = block->w();
+		double kernel[npoints];
```

Comment:
As I said before, this is a memory leak. This is also the cause of the build failures. Use something like `std::vector<double> kernel(npoints)` instead.